src/sql/magic_cmd.py

```diff
@@ -64,8 +73,19 @@ def _validate_execute_inputs(self, line):
             command, others = split[0].strip(), split[1:]
 
             if command in AVAILABLE_SQLCMD_COMMANDS:
-                if command not in ["profile"]:
+                if (
+                    command in COMMANDS_CONNECTION_REQUIRED
+                    and not ConnectionManager.current
+                ):
+                    raise exceptions.RuntimeError(
+                        f"Cannot execute %sqlcmd {command} because there "
+                        "is no active connection. Connect to a database "
+                        "and try again."
+                    )
+
+                if command in COMMANDS_SQLALCHEMY_ONLY:
                     util.support_only_sql_alchemy_connection("%sqlcmd")
```

Comment:
let's be more specific here and make the error message include which command isn't available, right now it always says `%sqlcmd`

Reply:
[fixed](https://github.com/ploomber/jupysql/pull/804/files#diff-2aa72d6ecafc2697ec8174353868dae816494131c70b7e48be6115fda9f61370R87)

Now it passes "%sqlcmd {command}" to `util.support_only_sql_alchemy_connection` so that error messages have that extract information